zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/authorization/AuthorizationStateTest.java

```diff
@@ -274,4 +274,271 @@ void shouldRemoveAllPermissionsByOwnerTypeAndID() {
+    final AuthorizationResourceType resourceType = AuthorizationResourceType.RESOURCE;
+    final Set<PermissionType> permissions = Set.of(PermissionType.CREATE, PermissionType.DELETE);
+    final var authorizationRecord =
+        new AuthorizationRecord()
+            .setAuthorizationKey(authorizationKey)
+            .setOwnerId(ownerId)
+            .setOwnerType(ownerType)
+            .setResourceId(resourceId)
+            .setResourceType(resourceType)
+            .setAuthorizationPermissions(permissions);
+
+    authorizationState.create(authorizationKey, authorizationRecord);
+
+    // when
+    final var updatedAuthorizationRecord =
+        new AuthorizationRecord()
+            .setAuthorizationKey(authorizationKey)
+            .setOwnerId("anotherOwnerId")
+            .setOwnerType(ownerType)
+            .setResourceId("anotherResourceId")
+            .setResourceType(resourceType)
+            .setAuthorizationPermissions(Set.of(PermissionType.READ));
+    authorizationState.update(authorizationKey, updatedAuthorizationRecord);
+
+    // then
+    final var persistedAuthorization = authorizationState.get(authorizationKey);
+    assertThat(persistedAuthorization).isPresent();
+    final var authorization = persistedAuthorization.get();
+    assertThat(authorization.getAuthorizationKey()).isEqualTo(authorizationKey);
+    assertThat(authorization.getOwnerId()).isEqualTo("anotherOwnerId");
+    assertThat(authorization.getOwnerType()).isEqualTo(ownerType);
+    assertThat(authorization.getResourceId()).isEqualTo("anotherResourceId");
+    assertThat(authorization.getResourceType()).isEqualTo(resourceType);
+    assertThat(authorization.getPermissions())
+        .containsExactlyInAnyOrderElementsOf(Set.of(PermissionType.READ));
+
+    final var resourceIdentifiers =
+        authorizationState.getResourceIdentifiers(
+            ownerType, "anotherOwnerId", resourceType, PermissionType.READ);
+    assertThat(resourceIdentifiers).containsExactly("anotherResourceId");
+
+    final var anotherResourceIdentifiers =
+        authorizationState.getResourceIdentifiers(
+            ownerType, "anotherOwnerId", resourceType, PermissionType.CREATE);
+    assertThat(anotherResourceIdentifiers).isEmpty();
+    final var anotherResourceIdentifiers2 =
+        authorizationState.getResourceIdentifiers(
+            ownerType, ownerId, resourceType, PermissionType.CREATE);
+    assertThat(anotherResourceIdentifiers2).isEmpty();
+    final var anotherResourceIdentifiers3 =
+        authorizationState.getResourceIdentifiers(
+            ownerType, ownerId, resourceType, PermissionType.READ);
+    assertThat(anotherResourceIdentifiers3).isEmpty();
+    final var anotherResourceIdentifiers4 =
+        authorizationState.getResourceIdentifiers(
+            ownerType, ownerId, resourceType, PermissionType.DELETE);
+    assertThat(anotherResourceIdentifiers4).isEmpty();
+  }
+
+  @Test
+  void shouldUpdateAuthorizationChangingPermission() {
```

Comment:
```suggestion
  void shouldUpdateAuthorizationChangingResourceType() {
```

I'd consider not updating the permissions here to keep the test case limited to one scenario.